OpenRA.Mods.Common/Activities/HarvesterDockSequence.cs

```diff
@@ -85,7 +102,7 @@ public override bool Tick(Actor self)
 
 					Harv.LastLinkedProc = Harv.LinkedProc;
 					Harv.LinkProc(self, null);
-					if (IsDragRequired)
+					if (IsDragRequired && wasDraggedToDock)
```

Comment:
`wasDraggedToDock` seems unnecessary since `IsDragRequired` can't change, right?

Reply:
The idea is to only play the reverse drag if the initial drag activity also ran, otherwise the actor will insta-jump to dragEnd and reverse-drag from there.

Reply:
As an alternative (not sure its better) we could make `IsDragRequired` not readonly and set it to false if the refinery is killed.

Reply:
If the initial drag activity doesn't run we should never reach this point, shouldn't we?

Reply:
https://github.com/OpenRA/OpenRA/pull/18737/files#diff-b3d67bffe7672e41778dd47afabf1ea99403a2542bf121c5ff4eb4cc82bf6904R64 cancels without going to another state.

Reply:
`Cancel()` unconditionally setting `dockingState = DockingState.Undock;` is the problem here. We should probably just drop that and add `IsCanceling` checks where necessary.